### PR TITLE
feat: Add Dropout3d layer to nn.layers

### PR DESCRIPTION
**Pull request by NripeshN**
## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Comment by NripeshN** [comment]
Hey @awni 
I am not sure if I should be adding tests here. I could not find tests for dropout2d and dropout.

**Comment by awni** [comment]
If you are up for adding a couple tests that would be fantastic! They can be pretty simple, like just check shapes are preserved, dtypes are preserved, we throw correctly on bad `p`, etc.

**Comment by NripeshN** [comment]
> If you are up for adding a couple tests that would be fantastic! They can be pretty simple, like just check shapes are preserved, dtypes are preserved, we throw correctly on bad `p`, etc.

Done!

File: ACKNOWLEDGMENTS.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ with a short description of your contribution(s) below. For example:
 
 MLX was developed with contributions from the following individuals:
 
+- Nripesh Niketan: Added `dropout3d` ops.
 - Nripesh Niketan: Added `softsign`, `softmax`, `hardswish`, `logsoftmax` activation functions.
```
Comment (awni) [inline ACKNOWLEDGMENTS.md:11]: I think it's simpler if you just append to your existing line rather than make a new line here (that's also what other have been doing!)

File: python/tests/test_nn.py [hunk 2]
```diff
@@ -792,6 +792,24 @@ def test_log_cosh_loss(self):
         loss = nn.losses.log_cosh_loss(inputs, targets, reduction="mean")
         self.assertAlmostEqual(loss.item(), 0.433781, places=6)
 
+    def test_dropout(self):
+        x = mx.ones((2, 4))
+        y = nn.Dropout(0.5)(x)
+        self.assertTrue(y.shape, x.shape)
+        self.assertTrue(y.dtype, mx.float32)
```
Comment (awni) [inline python/tests/test_nn.py:799]: Could you add a test where `x` is in `bfloat16` or `float16` as well (here and below?)

**Review by awni — CHANGES_REQUESTED** [review]
Looks great and thanks for adding the tests! Two minor comments, please address then we can merge!

File: ACKNOWLEDGMENTS.md (hunk 1 above)
Comment (NripeshN) [reply ACKNOWLEDGMENTS.md]: Done!

File: python/tests/test_nn.py (hunk 2 above)
Comment (NripeshN) [reply python/tests/test_nn.py:799]: Done!

**Review by awni — APPROVED** [review]
🚀  thanks!
